Neo4j.Driver/Neo4j.Driver/Internal/IO/BoltReader.cs

```diff
@@ -0,0 +1,279 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Neo4j.Driver.Internal.Messaging;
+using Neo4j.Driver.V1;
+using static Neo4j.Driver.Internal.IO.PackStream;
+
+namespace Neo4j.Driver.Internal.IO
+{
+    internal class BoltReader: IBoltReader
+    {
+        private readonly IChunkReader _chunkReader;
+        private readonly IPackStreamReader _packStreamReader;
+        private readonly ILogger _logger;
+        private readonly MemoryStream _bufferStream;
+
+        public BoltReader(Stream stream)
+            : this(stream, true)
+        {
+            
+        }
+
+        public BoltReader(Stream stream, bool supportBytes)
+            : this(stream, null, supportBytes)
+        {
+
+        }
+
+        public BoltReader(Stream stream, ILogger logger, bool supportBytes)
+            : this(new ChunkReader(stream, logger), logger, supportBytes)
+        {
+
+        }
+
+        public BoltReader(IChunkReader chunkReader, ILogger logger, bool supportBytes)
+        {
+            Throw.ArgumentNullException.IfNull(chunkReader, nameof(chunkReader));
+
+            _logger = logger;
+            _chunkReader = chunkReader;
+            _bufferStream = new MemoryStream();
+            _packStreamReader = supportBytes ? new PackStreamReader(_bufferStream) : new PackStreamReaderBytesIncompatible(_bufferStream);
+        }
+
+
+
+        public void Read(IMessageResponseHandler responseHandler)
+        {
+            _bufferStream.SetLength(0);
+
+            _chunkReader.ReadNextChunk(_bufferStream);
```

Comment:
If I got the code here correctly, the whole logic depends on the assumption that a whole message (such as a Record message) could be put into `_bufferStream`. I do not think that we could assume on that.

Pls correct me if I understand this part wrongly.